cylc/flow/data_store_mgr.py

```diff
@@ -330,6 +341,9 @@ def apply_delta(key, delta, data):
                         'on update application: %s' % str(exc)
                     )
                     continue
+    # Clear memory accumulation
+    if key == WORKFLOW:
+        data[key] = reset_protobuf_object(MESSAGE_MAP[key], data[key])
```

Comment:
Why do we only do this for WORKFLOW fields, not all fields?

Reply:
Because `PbWorkflow` the only long-lived field, whereas nodes/edges/etc will cause a little accumulation but usually have finite updates and existence (are pruned).. See:
https://github.com/cylc/cylc-uiserver/issues/682#issuecomment-2800872753

Reply:
So updating **any** repeated field in protobuf causes a memory leak, but we are only really concerned by the workflow field due to its longevity?

Reply:
Correct, "memory leak" for the most part on nodes would be changes in state or other fields.. But these typically only have a finite number of updates done to them, and are ultimately deleted..

The only other object I can think of is the "mean_elapse_time" on the `PbTask` objects... 
(now reset)